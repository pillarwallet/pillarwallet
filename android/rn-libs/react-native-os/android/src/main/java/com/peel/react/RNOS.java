/**
 * Copyright (c) 2015-present, Peel Technologies, Inc.
 * All rights reserved.
 */

package com.peel.react.rnos;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.net.ConnectivityManager;
import android.os.Bundle;

import com.facebook.react.bridge.Arguments;
import com.facebook.react.bridge.LifecycleEventListener;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.modules.core.DeviceEventManagerModule;

import java.net.Inet4Address;
import java.net.Inet6Address;
import java.net.InetAddress;
import java.net.InterfaceAddress;
import java.net.NetworkInterface;
import java.net.SocketException;
import java.util.Arrays;
import java.util.Enumeration;
import java.util.HashMap;
import java.util.Map;


public final class RNOS extends ReactContextBaseJavaModule implements LifecycleEventListener {
    /* package */ static final String TAG = "RNOS";
    /* package */ final ConnectivityManager mConnectivityManager;
    /* package */ final ConnectivityBroadcastReceiver mConnectivityBroadcastReceiver;

    public RNOS(ReactApplicationContext reactContext) {
        super(reactContext);
        mConnectivityManager =
                (ConnectivityManager) reactContext.getSystemService(Context.CONNECTIVITY_SERVICE);
        mConnectivityBroadcastReceiver = new ConnectivityBroadcastReceiver();

        reactContext.addLifecycleEventListener(this);
    }

    @Override
    public String getName() {
        return TAG;
    }

    @Override
    public Map<String, Object> getConstants() {
        // set constants as initial values
        final Map<String, Object> constants = new HashMap<>();

        final Map<String, Object> networkMap = new HashMap<>();
        try {
            // extract bundle to map
            final Bundle networkInterfaces = getNetworkInterfaces();
            for (String key : networkInterfaces.keySet()) {
                networkMap.put(key, networkInterfaces.get(key));
            }
        } catch (SocketException e) {
            e.printStackTrace();
        }

        constants.put("networkInterfaces", networkMap);
        constants.put("homedir", getReactApplicationContext().getApplicationInfo().dataDir);
        return constants;
    }

    @Override
    public void onHostResume() {
        registerReceiver();
    }

    @Override
    public void onHostPause() {
        unregisterReceiver();
    }

    @Override
    public void onHostDestroy() {
        unregisterReceiver();
    }

    private void registerReceiver() {
        IntentFilter filter = new IntentFilter();
        filter.addAction(ConnectivityManager.CONNECTIVITY_ACTION);
        getReactApplicationContext().registerReceiver(mConnectivityBroadcastReceiver, filter);
        mConnectivityBroadcastReceiver.setRegistered(true);
    }

    private void unregisterReceiver() {
        if (mConnectivityBroadcastReceiver.isRegistered()) {
            getReactApplicationContext().unregisterReceiver(mConnectivityBroadcastReceiver);
            mConnectivityBroadcastReceiver.setRegistered(false);
        }
    }

    public void updateAndSendOsInfo() {
        Bundle osInfo = new Bundle();

        try {
            final Bundle networkInterfaces = getNetworkInterfaces();
            osInfo.putBundle("networkInterfaces", networkInterfaces);
        } catch (SocketException e) {
            e.printStackTrace();
        }

        final ReactContext reactContext = getReactApplicationContext();
        reactContext
                .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
                .emit("rn-os-info", Arguments.fromBundle(osInfo));
    }

    private Bundle getNetworkInterfaces() throws SocketException {
        Bundle ifaces = new Bundle();
        Enumeration<NetworkInterface> list = NetworkInterface.getNetworkInterfaces();
        while (list.hasMoreElements()) {
            final NetworkInterface iface = list.nextElement();
            String mac = "00:00:00:00:00:00";
            boolean internal = true;

            try {
                internal = iface.isLoopback();
            } catch (SocketException se) {
                // keep calm and query on
                se.printStackTrace();
            }

            try {
                final byte[] macBytes = iface.getHardwareAddress();

                if (macBytes != null) {
                    final StringBuilder sb = new StringBuilder();
                    for (int i = 0; i < macBytes.length; i++) {
                        sb.append(String.format("%02X%s", macBytes[i], (i < macBytes.length - 1) ? ":" : ""));
                    }
                    mac = sb.toString();
                }
            } catch (SocketException se) {
                // keep calm and query on
                se.printStackTrace();
            }

            for (InterfaceAddress address : iface.getInterfaceAddresses()) {
                Bundle ifaceInfo = new Bundle();

                ifaceInfo.putBoolean("internal", internal);
                ifaceInfo.putString("mac", mac);

                InetAddress inet = address.getAddress();

                // getHostAddress adds %scope_id for ipv6
                String hostAddress = inet.getHostAddress();
                int end = hostAddress.indexOf("%");
                if (end > -1) {
                    hostAddress = hostAddress.substring(0, end);
                }
                ifaceInfo.putString("address", hostAddress);
                if (inet instanceof Inet6Address) {
                    ifaceInfo.putString("family", "IPv6");
                    ifaceInfo.putInt("scopeid", ((Inet6Address) inet).getScopeId());
                } else {
                    ifaceInfo.putString("family", "IPv4");
                }

                String netmask = "";
                short prefixLength = address.getNetworkPrefixLength();
                if (inet instanceof Inet4Address) {
                    final int value = 0xffffffff << (32 - prefixLength);
                    netmask = String.format("%d.%d.%d.%d", (value >> 24) & 0xFF,
                            (value >> 16) & 0xFF,  (value >> 8) & 0xFF, value & 0xFF);
                } else {
                    final long[] value = new long[] { 0xffffffffffffffffl, 0xffffffffffffffffl };
                    if (prefixLength <= 64) {
                        value[1] = value[1] << (64 - prefixLength);
                    } else {
                        value[1] = 0;
                        value[0] = value[0] << (64 - (prefixLength - 64));
                    }

                    for (long crtLong : value) { //for every long: it should be two of them
                        for (int i = 0; i < 4; i++) { //we display in total 4 parts for every long
                            netmask += (netmask.length() == 0 ? "" : ":") +
                                ((crtLong & 0xFFFF) == 0 ? "" : Long.toHexString(crtLong & 0xFFFF));
                            crtLong = crtLong >> 16;
                        }
                    }
                }

                ifaceInfo.putString("netmask", netmask);

                Bundle[] bundles = (Bundle[]) ifaces.getParcelableArray(iface.getDisplayName());
                if (bundles == null) {
                    bundles = new Bundle[] { ifaceInfo };
                } else {
                    Bundle[] tmp = Arrays.copyOf(bundles, bundles.length + 1);
                    tmp[bundles.length] = ifaceInfo;
                    bundles = tmp;
                }

                ifaces.putParcelableArray(iface.getDisplayName(), bundles);
            }
        }

        return ifaces;
    }

    /**
     * Class that receives intents whenever the connection type changes.
     * NB: It is possible on some devices to receive certain connection type changes multiple times.
     */
    private class ConnectivityBroadcastReceiver extends BroadcastReceiver {

        //TODO: Remove registered check when source of crash is found. t9846865
        private boolean isRegistered = false;

        public void setRegistered(boolean registered) {
            isRegistered = registered;
        }

        public boolean isRegistered() {
            return isRegistered;
        }

        @Override
        public void onReceive(Context context, Intent intent) {
            if (intent.getAction().equals(ConnectivityManager.CONNECTIVITY_ACTION)) {
                updateAndSendOsInfo();
            }
        }
    }
}
