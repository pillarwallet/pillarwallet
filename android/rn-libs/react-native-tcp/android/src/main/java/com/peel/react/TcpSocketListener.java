package com.peel.react;

import java.net.InetSocketAddress;

/**
 * Created by aprock on 12/28/15.
 */
public interface TcpSocketListener {
    // server
    void onConnection(Integer serverId, Integer clientId, InetSocketAddress socketAddress);

    // client and server
    void onConnect(Integer id, InetSocketAddress socketAddress);
    void onData(Integer id, byte[] data);
    void onClose(Integer id, String error);
    void onError(Integer id, String error);
}
