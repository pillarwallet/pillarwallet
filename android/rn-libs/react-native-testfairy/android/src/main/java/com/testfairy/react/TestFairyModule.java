package com.testfairy.react;

import android.app.Activity;
import android.util.Log;
import android.view.View;

import com.facebook.react.bridge.Callback;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.ReadableArray;
import com.facebook.react.bridge.ReadableMap;
import com.facebook.react.bridge.ReadableMapKeySetIterator;
import com.facebook.react.bridge.ReadableType;
import com.facebook.react.bridge.UiThreadUtil;

import com.testfairy.TestFairy;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

public class TestFairyModule extends ReactContextBaseJavaModule {

    public TestFairyModule(ReactApplicationContext reactContext) {
        super(reactContext);
    }

    @Override
    public String getName() {
        return "TestFairyBridge";
    }

    @ReactMethod
    public void begin(final String appKey, ReadableMap map) {
        runOnUi(new Runnable() {
            @Override
            public void run() {
                TestFairy.begin(getReactApplicationContext(), appKey);
            }
        });
    }

    @ReactMethod
    public void setCorrelationId(final String correlationId) {
        runOnUi(new Runnable() {
            @Override
            public void run() {
                TestFairy.setCorrelationId(correlationId);
            }
        });
    }

    @ReactMethod
    public void identify(final String identity, final ReadableMap map) {
        runOnUi(new Runnable() {
            @Override
            public void run() {
                if (map == null) {
                    TestFairy.identify(identity, null);
                } else {
                    final Map<String, Object> traits = convertMap(map);
                    TestFairy.identify(identity, traits);
                }
            }
        });
    }

    @ReactMethod
    public void takeScreenshot() {
        Log.i("TestFairyModule", "Android does not support taking screen shots");
    }

    @ReactMethod
    public void pause() {
        runOnUi(new Runnable() {
            @Override
            public void run() {
                TestFairy.pause();
            }
        });
    }

    @ReactMethod
    public void pushFeedbackController() {
        runOnUi(new Runnable() {
            @Override
            public void run() {
                TestFairy.showFeedbackForm();
            }
        });
    }

    @ReactMethod
    public void resume() {
        runOnUi(new Runnable() {
            @Override
            public void run() {
                TestFairy.resume();
            }
        });
    }

    @ReactMethod
    public void checkpoint(final String checkpoint) {
        runOnUi(new Runnable() {
            @Override
            public void run() {
                TestFairy.addEvent(checkpoint);
            }
        });
    }

    @ReactMethod
    public void hideWebViewElements(final String cssSelector) {
        runOnUi(new Runnable() {
            @Override
            public void run() {
                Log.i("TestFairyModule", "Android does not support hiding web view elements");
            }
        });
    }

    @ReactMethod
    public void sendUserFeedback(final String feedback) {
        runOnUi(new Runnable() {
            @Override
            public void run() {
                TestFairy.sendUserFeedback(feedback);
            }
        });
    }

    @ReactMethod
    public void setServerEndpoint(final String url) {
        runOnUi(new Runnable() {
            @Override
            public void run() {
                TestFairy.setServerEndpoint(url);
            }
        });
    }

    @ReactMethod
    public void log(final String message) {
        runOnUi(new Runnable() {
            @Override
            public void run() {
                TestFairy.log("TFReactNative", message);
            }
        });
    }

    @ReactMethod
    public void sessionUrl(final Callback callback) {
        runOnUi(new Runnable() {
            @Override
            public void run() {
                callback.invoke(TestFairy.getSessionUrl());
            }
        });
    }

    @ReactMethod
    public void version(final Callback callback) {
        runOnUi(new Runnable() {
            @Override
            public void run() {
                callback.invoke(TestFairy.getSessionUrl());
            }
        });
    }

    @ReactMethod
    public void setScreenName(final String name) {
        runOnUi(new Runnable() {
            @Override
            public void run() {
                TestFairy.setScreenName(name);
            }
        });
    }

    @ReactMethod
    public void stop() {
        runOnUi(new Runnable() {
            @Override
            public void run() {
                TestFairy.stop();
            }
        });
    }

    @ReactMethod
    public void setUserId(final String userId) {
        runOnUi(new Runnable() {
            @Override
            public void run() {
                TestFairy.setUserId(userId);
            }
        });
    }

    @ReactMethod
    public void setAttribute(final String key, final String value) {
        runOnUi(new Runnable() {
            @Override
            public void run() {
                TestFairy.setAttribute(key, value);
            }
        });
    }

    @ReactMethod
    public void hideView(final int tag) {
        runOnUi(new Runnable() {
            @Override
            public void run() {
                TestFairy.hideView(tag);
            }
        });
    }

		@ReactMethod
    public void enableCrashHandler() {
			runOnUi(new Runnable() {
					@Override
					public void run() {
							TestFairy.enableCrashHandler();
					}
			});
		}

		@ReactMethod
    public void disableCrashHandler() {
			runOnUi(new Runnable() {
					@Override
					public void run() {
							TestFairy.disableCrashHandler();
					}
			});
		}

		@ReactMethod
    public void enableMetric(final String metric) {
			runOnUi(new Runnable() {
					@Override
					public void run() {
							TestFairy.enableMetric(metric);
					}
			});
		}

		@ReactMethod
    public void disableMetric(final String metric) {
			runOnUi(new Runnable() {
					@Override
					public void run() {
							TestFairy.disableMetric(metric);
					}
			});
		}

		@ReactMethod
    public void enableVideo(final String policy, final String quality, final float framesPerSecond) {
			runOnUi(new Runnable() {
					@Override
					public void run() {
							TestFairy.enableVideo(policy, quality, framesPerSecond);
					}
			});
		}

		@ReactMethod
    public void disableVideo() {
			runOnUi(new Runnable() {
					@Override
					public void run() {
							TestFairy.disableVideo();
					}
			});
		}

		@ReactMethod
    public void enableFeedbackForm(final String method) {
			runOnUi(new Runnable() {
					@Override
					public void run() {
							TestFairy.enableFeedbackForm(method);
					}
			});
		}

		@ReactMethod
    public void disableFeedbackForm() {
			runOnUi(new Runnable() {
					@Override
					public void run() {
							TestFairy.disableFeedbackForm();
					}
			});
		}

		@ReactMethod
    public void setMaxSessionLength(final float seconds) {
			runOnUi(new Runnable() {
					@Override
					public void run() {
							TestFairy.setMaxSessionLength(seconds);
					}
			});
		}

    private Map<String, Object> convertMap(ReadableMap map) {
        Map<String, Object> input = new HashMap<String, Object>();
        ReadableMapKeySetIterator iterator = map.keySetIterator();
        while (iterator.hasNextKey()) {
            String key = iterator.nextKey();
            ReadableType type = map.getType(key);
            switch (type) {
                case Boolean:
                    input.put(key, map.getBoolean(key));
                    break;
                case String:
                    input.put(key, map.getString(key));
                    break;
                case Number:
                    input.put(key, map.getDouble(key));
                    break;
                case Array:
                    input.put(key, convertArray(map.getArray(key)));
                    break;
                case Map:
                    input.put(key, convertMap(map.getMap(key)));
                default:
                    break;
            }
        }

        return input;
    }

    private ArrayList<Object> convertArray(ReadableArray array) {
        ArrayList<Object> input = new ArrayList<Object>();
        ReadableType singleType = null;
        for (int index = 0; index < array.size(); index++) {
            ReadableType type = array.getType(index);
            if (singleType == null)
                singleType = type;

            if (type != singleType) {
                Log.d("TestFairyModule", "Cannot mix types in array objects expecting type [" + singleType + "] found [" + type + "] in array. Skipping");
                continue;
            }

            switch (type) {
                case Boolean:
                    input.add(array.getBoolean(index));
                    break;
                case String:
                    input.add(array.getString(index));
                    break;
                case Number:
                    input.add(array.getDouble(index));
                    break;
                case Array:
                    input.add(convertArray(array.getArray(index)));
                    break;
                case Map:
                    input.add(convertMap(array.getMap(index)));
                default:
                    break;
            }
        }

        return input;
    }

    private void runOnUi(Runnable runnable) {
        UiThreadUtil.runOnUiThread(runnable);
    }
}
