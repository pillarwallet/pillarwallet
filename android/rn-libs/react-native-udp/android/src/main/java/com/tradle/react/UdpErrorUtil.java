/**
 *  UdpErrorUtil.java
 *  react-native-udp
 *
 *  Created by Andy Prock on 9/24/15.
 */

package com.tradle.react;

import javax.annotation.Nullable;

import com.facebook.react.bridge.Arguments;
import com.facebook.react.bridge.WritableMap;

/**
 * Helper class for udp errors.
 */
public class UdpErrorUtil {
    /**
     * Create Error object to be passed back to the JS callback.
     */
    /* package */ static WritableMap getError(@Nullable String key, String errorMessage) {
        WritableMap errorMap = Arguments.createMap();
        errorMap.putString("message", errorMessage);
        if (key != null) {
            errorMap.putString("key", key);
        }
        return errorMap;
    }
}
