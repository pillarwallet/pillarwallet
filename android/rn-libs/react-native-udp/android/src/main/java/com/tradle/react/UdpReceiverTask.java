/**
 *  UdpReceiverTask.java
 *  react-native-udp
 *
 *  Created by Andy Prock on 9/24/15.
 */

package com.tradle.react;

import android.os.AsyncTask;
import android.util.Base64;
import android.util.Pair;

import java.io.IOException;
import java.net.DatagramPacket;
import java.net.DatagramSocket;
import java.net.InetAddress;

/**
 * This is a specialized AsyncTask that receives data from a socket in the background, and
 * notifies it's listener when data is received.  This is not threadsafe, the listener
 * should handle synchronicity.
 */
public class UdpReceiverTask extends AsyncTask<Pair<DatagramSocket, UdpReceiverTask.OnDataReceivedListener>, Void, Void> {
    private static final String TAG = "UdpReceiverTask";
    private static final int MAX_UDP_DATAGRAM_LEN = 1024;

    /**
     * An infinite loop to block and read data from the socket.
     */
    @Override
    protected Void doInBackground(Pair<DatagramSocket, UdpReceiverTask.OnDataReceivedListener>... params) {
        if (params.length > 1) {
            throw new IllegalArgumentException("This task is only for a single socket/listener pair.");
        }

        DatagramSocket socket = params[0].first;
        OnDataReceivedListener receiverListener = params[0].second;

        final byte[] buffer = new byte[MAX_UDP_DATAGRAM_LEN];
        DatagramPacket packet = new DatagramPacket(buffer, buffer.length);

        while (!isCancelled()) {
            try {
                socket.receive(packet);

                final InetAddress address = packet.getAddress();
                final String base64Data = Base64.encodeToString(packet.getData(), packet.getOffset(),
                    packet.getLength(), Base64.NO_WRAP);
                receiverListener.didReceiveData(base64Data, address.getHostName(), packet.getPort());
            } catch (IOException ioe) {
                if (receiverListener != null) {
                    receiverListener.didReceiveError(ioe.getMessage());
                }
                this.cancel(false);
            } catch (RuntimeException rte) {
                if (receiverListener != null) {
                    receiverListener.didReceiveRuntimeException(rte);
                }
                this.cancel(false);
            }
        }

        return null;
    }

    /**
     * Listener interface for receive events.
     */
    public interface OnDataReceivedListener {
        void didReceiveData(String data, String host, int port);
        void didReceiveError(String message);
        void didReceiveRuntimeException(RuntimeException exception);
    }
}
