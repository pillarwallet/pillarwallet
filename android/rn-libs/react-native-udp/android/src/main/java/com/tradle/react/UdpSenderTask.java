/**
 *  UdpSenderTask.java
 *  react-native-udp
 *
 *  Created by Andy Prock on 9/24/15.
 */

package com.tradle.react;

import android.os.AsyncTask;

import java.io.IOException;
import java.lang.ref.WeakReference;
import java.net.DatagramPacket;
import java.net.DatagramSocket;
import java.net.SocketAddress;

/**
 * Specialized AsyncTask that transmits data in the background, and notifies listeners of the result.
 */
public class UdpSenderTask extends AsyncTask<UdpSenderTask.SenderPacket, Void, Void> {
    private static final String TAG = "UdpSenderTask";

    private DatagramSocket mSocket;
    private WeakReference<OnDataSentListener> mListener;

    public UdpSenderTask(DatagramSocket socket, OnDataSentListener listener) {
        this.mSocket = socket;
        this.mListener = new WeakReference<OnDataSentListener>(listener);
    }

    @Override
    protected Void doInBackground(SenderPacket... params) {
        OnDataSentListener listener = mListener.get();

        try {
            SenderPacket packet = params[0];
            mSocket.send(new DatagramPacket(packet.data, packet.data.length, packet.socketAddress));

            if (listener != null) {
                listener.onDataSent(this);
            }
        } catch (IOException e) {
            if (listener != null) {
                listener.onDataSentError(this, e.getMessage());
            }
        } catch (RuntimeException rte) {
            if (listener != null) {
                listener.onDataSentRuntimeException(this, rte);
            }
        }

        return null;
    }

    /**
     * Simple class to marshall outgoing data across to this AsyncTask
     */
    public static class SenderPacket {
        SocketAddress socketAddress;
        byte[] data;
    }

    /**
     * Callbacks for data send events.
     */
    public interface OnDataSentListener {
        void onDataSent(UdpSenderTask task);
        void onDataSentError(UdpSenderTask task, String error);
        void onDataSentRuntimeException(UdpSenderTask task, RuntimeException exception);
    }
}
