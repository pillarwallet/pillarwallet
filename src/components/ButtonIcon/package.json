{
  "name": "ButtonIcon.js",
  "version": "0.0.0",
  "private": true,
  "main": "./ButtonIcon.js"
}
