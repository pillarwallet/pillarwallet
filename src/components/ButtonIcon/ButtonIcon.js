// @flow
import * as React from 'react';
import { StyleSheet } from 'react-native';
import { Icon, Button } from 'native-base';

type Props = {
  icon: string,
  color?: string,
  onPress?: Function,
  fontSize?: number,
  style?: StyleSheet.Styles,
}

const ButtonIcon = (props: Props) => {
  const {
    icon,
    onPress,
    fontSize,
    color,
    style,
  } = props;
  return (
    <Button transparent iconLeft={0} onPress={onPress} style={style}>
      <Icon active={false} name={icon} style={{ fontSize, color }} />
    </Button>
  );
};

export default ButtonIcon;
