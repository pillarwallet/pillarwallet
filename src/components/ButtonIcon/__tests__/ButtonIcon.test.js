// @flow
import * as React from 'react';
import renderer from 'react-test-renderer';
import ButtonIcon from '../ButtonIcon';

describe('Button Icon', () => {
  it('should render Button Icon correctly', () => {
    const component = renderer.create(<ButtonIcon icon="settings" />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
