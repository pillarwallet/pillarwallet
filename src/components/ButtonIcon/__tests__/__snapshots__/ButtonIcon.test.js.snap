// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Icon should render Button Icon correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "transparent",
      "borderBottomWidth": null,
      "borderColor": "#007aff",
      "borderLeftWidth": null,
      "borderRadius": 5,
      "borderRightWidth": null,
      "borderTopWidth": null,
      "elevation": 0,
      "flexDirection": "row",
      "height": 45,
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingBottom": 6,
      "paddingTop": 6,
      "shadowColor": null,
      "shadowOffset": null,
      "shadowOpacity": null,
      "shadowRadius": null,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    active={false}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Array [
          Object {
            "color": "#000",
            "fontSize": 30,
          },
          Object {
            "color": "#007aff",
            "fontSize": 24,
            "marginLeft": 16,
            "marginRight": 16,
            "paddingTop": 2,
          },
          Object {
            "color": undefined,
            "fontSize": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
          },
        ],
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
    type="Ionicons"
    virtual={undefined}
  >
    
  </Text>
</View>
`;
